runelite-client/src/main/java/net/runelite/client/plugins/blastmine/BlastMineRockOverlay.java

```diff
@@ -0,0 +1,209 @@
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.blastmine;
+
+import com.google.common.collect.Sets;
+import net.runelite.api.Client;
+import net.runelite.api.GameObject;
+import net.runelite.api.ItemID;
+import net.runelite.api.ObjectID;
+import net.runelite.api.Perspective;
+import net.runelite.api.Tile;
+import net.runelite.api.coords.LocalPoint;
+import net.runelite.api.widgets.Widget;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import java.awt.BasicStroke;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.Polygon;
+import java.awt.geom.Arc2D;
+import java.awt.image.BufferedImage;
+import java.util.Collections;
+import java.util.Set;
+
+public class BlastMineRockOverlay extends Overlay
+{
+	private static final int TIMER_SIZE = 25;
+	private static final int TIMER_BORDER_WIDTH = 1;
+	private static final int MAX_DISTANCE = 16;
+	private static final int WARNING_DISTANCE = 2;
+	private static final Set<Integer> WALL_OBJECTS = Collections.unmodifiableSet(Sets.newHashSet(ObjectID.NULL_28570, ObjectID.NULL_28571, ObjectID.NULL_28572, ObjectID.NULL_28573, ObjectID.NULL_28574,
+			ObjectID.NULL_28575, ObjectID.NULL_28576, ObjectID.NULL_28577, ObjectID.NULL_28578, ObjectID.HARD_ROCK, ObjectID.HARD_ROCK_28580,
+			ObjectID.CAVITY, ObjectID.CAVITY_28582, ObjectID.POT_OF_DYNAMITE, ObjectID.POT_OF_DYNAMITE_28584, ObjectID.POT_OF_DYNAMITE_28585,
+			ObjectID.POT_OF_DYNAMITE_28586, ObjectID.SHATTERED_ROCKFACE, ObjectID.SHATTERED_ROCKFACE_28588));
+
+	private final Client client;
+	private final BlastMinePlugin plugin;
+	private final BlastMinePluginConfig config;
+
+	private Color timerColor;
+	private Color warningColor;
+
+	@Inject
+	private ItemManager itemManager;
+
+	@Inject
+	BlastMineRockOverlay(@Nullable Client client, BlastMinePlugin plugin, BlastMinePluginConfig config)
+	{
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.ABOVE_SCENE);
+		this.plugin = plugin;
+		this.config = config;
+		this.client = client;
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		plugin.getRocks().removeIf(rock -> (rock.getRemainingTimeRelative() == 1 && rock.getType() != BlastMineRockType.NORMAL) || (rock.getRemainingFuseTimeRelative() == 1 && rock.getType() == BlastMineRockType.LIT));
+		drawRocks(graphics);
+		return null;
+	}
+
+	public void updateColors()
+	{
+		timerColor = config.getTimerColor();
+		warningColor = config.getWarningColor();
+	}
+
+	private void drawRocks(Graphics2D graphics)
+	{
+		Tile[][][] tiles = client.getRegion().getTiles();
+		BufferedImage chiselIcon = itemManager.getImage(ItemID.CHISEL);
```

Comment:
Can't you store these images beforehand? I don't think getting them every frame is good idea.